protocol/x/clob/e2e/conditional_orders_test.go

```diff
@@ -958,6 +958,1125 @@ func TestConditionalOrder(t *testing.T) {
 	}
 }
 
+func TestConditionalOrder_TriggeringUsingMatchedPrice(t *testing.T) {
```

Comment:
mostly replicated from the above e2e tests. 

only difference is that instead of price updates, updated to use match price.